frontend/packages/knative-plugin/src/components/add/EventSource.tsx

```diff
@@ -56,10 +66,15 @@ const EventSource: React.FC<Props> = ({
     data: {
       [typeEventSource.toLowerCase()]: getEventSourceData(typeEventSource.toLowerCase()),
     },
+    yamlData: !_.isEmpty(eventYaml) ? safeDump(eventYaml) : '',
+    editorType: EditorType.Form,
   };
 
   const createResources = (rawFormData: any): Promise<K8sResourceKind> => {
-    const knEventSourceResource = getEventSourcesDepResource(rawFormData);
+    let knEventSourceResource = getEventSourcesDepResource(rawFormData);
+    if (rawFormData.editorType === EditorType.YAML) {
+      knEventSourceResource = getEventSourcesResource(rawFormData);
```

Comment:
I am not sure what is the difference between `getEventSourcesResource` and `getEventSourcesDepResource` going by just the name of the functions. Better to name it in a way that gives more info upfront.

Reply:
changed the function name to `getEventSourcesResourceFromYAML`